src/test/java/org/apache/commons/io/input/QueueInputStreamTest.java

```diff
@@ -123,14 +129,53 @@ public void testAvailableAfterOpen(final String inputData) throws IOException {
     @MethodSource("inputData")
     public void testBufferedReads(final String inputData) throws IOException {
         final BlockingQueue<Integer> queue = new LinkedBlockingQueue<>();
-        try (BufferedInputStream inputStream = new BufferedInputStream(new QueueInputStream(queue));
-                QueueOutputStream outputStream = new QueueOutputStream(queue)) {
+        try (BufferedInputStream inputStream = new BufferedInputStream(new QueueInputStream(queue)); QueueOutputStream outputStream = new QueueOutputStream(queue)) {
             outputStream.write(inputData.getBytes(StandardCharsets.UTF_8));
             final String actualData = IOUtils.toString(inputStream, StandardCharsets.UTF_8);
             assertEquals(inputData, actualData);
         }
     }
 
+    @ParameterizedTest(name = "inputData={0}")
+    @MethodSource("inputData")
+    public void testReadLineByLineQueue(final String inputData) throws IOException {
+        final String[] lines = inputData.split("\n");
+        final BlockingQueue<Integer> queue = new LinkedBlockingQueue<>();
+        try (InputStream inputStream = QueueInputStream.builder().setBlockingQueue(queue)
+                                                       .setTimeout(Duration.ofHours(1))
+                                                       .get();
+             QueueOutputStream outputStream = new QueueOutputStream(queue)) {
```

Comment:
_Nitpick_: by using `QueueInputStream.newQueueOutputStream()` we gain one more line of coverage:

```suggestion
        try (QueueInputStream inputStream = QueueInputStream.builder().setBlockingQueue(queue)
                                                            .setTimeout(Duration.ofHours(1))
                                                            .get();
             OutputStream outputStream = inputStream.newQueueOutputStream()) {
```

Reply:
Fixed in e91989163f7a5e856ae1b168b00c14ef49b8d8fe